### DISCUSSION TITLE
Closing a relay

### DISCUSSION THREAD

**Discussion by dariusc93** [libp2p/rust-libp2p#3430] [Q&A]
Doing some of my testing with relays, I begun to notice a relay connection is not closing. If we want to close a connection to `A` circuit,  `Swarm::remove_listener` is used but recurring calls to the function with the same listener id would return true as if its not closing. ~~It also looks like the relay is also still be reported by identify too.~~ Is there a proper way to close (or stop listening) to the relay if say we want to use another one without holding reservation to an old relay, or if upnp ends up being used? 

EDIT: Ill check the debug logs and post them shortly

**Comment by thomaseizinger** [comment]
The connection should be closed once it idles. Which behaviours are you composing?
